### ISSUE TITLE
Sign your .onion address with your PGP Private key and Display the signed onion address on get hub in your clearnet website

### ISSUE THREAD

**Issue by Baccount** [micahflee/onionshare#555]
Sign your .onion address with your PGP Private key and Display the signed onion address on Github  your clearnet website. This will allow users to verify that you're V3 onion website is correct. This would remove the need for a Sha-256 hash for the compiled Onionshare. This should be implemented after #523 is complete. Your PGP public key should also be displayed on both Github and your website.
V3 onion sites should prevent any MITM attacks or Forgery of the onion site.

V3 Onion services.
1.**Better crypto (replaced SHA1/DH/RSA1024 with SHA3/ed25519/curve25519)**
2.**Improved directory protocol, leaking much less information to directory servers.**
4.**Better onion address security against impersonation.**
[https://trac.torproject.org](https://trac.torproject.org/projects/tor/wiki/doc/NextGenOnions?version=50#IntrotoNextGenOnionServicesakaprop224)
@micahflee

**Comment by Baccount** [comment]
Example of what it looks like.
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

http://lldan5gahapx5k7iafb3s4ikijc4ni7gx5iywdflkba5y2ezyg6sjgyd.onion/ 
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCAAdFiEEcdl8qnQpTsO5v66SBxtSP3cpNREFAlphOBUACgkQBxtSP3cp
NRF0sQv/ThyThHIB3TCF8qcmy0ryvyy9ZZRssbwZfStXztBRWkghafXckBqV+0Co
QeMvYfTfoOzkTZK0YkN8/JAuzuBuWgPco2eCY8yxwtoOr4VZdFiuTTK8vCJUL7U5
ae57ZZoHpR0Wazqkdmf8iMGvtpnl8UJAPkf4Uyebm88/yNQ05fd//iDXSY99Ga7e
24LWOmhbu3M14a8UGoRvqZBTcj3hddznc3CB1yEYSWE3gWVI2nn7vfTSR3v+ljjl
XiW4xsn91Y+fTT+h0mIpAQwOL9uWZKo+a9azl/24t/VYXeAhudtQSXMgHBeuHQ3v
JLO7LZ6/seCHI/ExCZZngDKsJ/AQqBFvVtV+ygmMe5kPBkWCaF0TUf1LSdICr4+q
EjW0WR8iZwRQTJ7xRXb1fqZe3gQB5QyYuA7wjNR1rvKbuRZ9zQWklQMmwS9YNvtL
yaVHfGwhQEFkn/RYkZsMRzcmuf/YHbW1QIrLFqFKcVX/VjhhV96XCwq5YqNwS5+Z
wEvB2NMb
=BScP
-----END PGP SIGNATURE-----

```

**Comment by mig5** [comment]
@micahflee this seems quick and easy to do now on the onionshare.org infrastructure.

It should be noted that the v3 address can already be found in the git repo as a signed Git commit because we put it in the Update Check logic: https://github.com/micahflee/onionshare/commit/8cc651a370f4925bf94001e89028968d7c8c790a#diff-77947f8d1e23075111b090a905fb84d0

However, because I introduced the change, it's my signed git commit and not Micah's, so the idea above is probably still better. Or we could both sign such a file..

**Comment by mig5** [comment]
@micahflee perhaps where the onion addresses are printed in the footer, add in parenthesis a hyperlink '(proof)' ?

**Comment by micahflee** [comment]
Ok, renamed it to `onionshare.txt` and actually deployed the new website too: https://onionshare.org/onion.txt

I was considering something like that but I thought it might deserve a bit more explanation... like at least saying what key it's signed with or something. Although the onion address itself doesn't have any explanation right now, so why not do that now but maybe write in more detail in the wiki.

**Comment by micahflee** [comment]
![screenshot_2018-09-15_21-44-09](https://user-images.githubusercontent.com/156128/45592995-870ac980-b930-11e8-9b40-0aa2742310a8.png)

**Comment by mig5** [comment]
Yes I agree, we should accompany this with a wiki page. Just nice to have it clickable on the site.

Also +1 for removing the v2 onion address on the site, I think that's fine, Whonix has done essentially the same too.
